packages/inventory-compliance/src/SystemRulesTable/SystemRulesTable.js

```diff
@@ -182,8 +185,10 @@ class SystemRulesTable extends React.Component {
     getRules = () => {
         const rules = toRulesArray(this.props.profileRules);
         return this.sortedRules(
-            this.filteredRules(
-                this.selectedRules(this.openedRules(rules))
+            this.selectedRules(
+                this.filteredRules(
```

Comment:
What's the aim of the inversion of method calls?
Shouldn't `selectedRules` be called before filtering?

IIRC `selectedRules` sets the selected boolean.

The filtering does not work for me as expected. It always yields 0 results even if there are some rules selected. Could this be the reason?

Reply:
`selectedRules` was renamed to `addIsSelected`. As you've said, `addIsSelected` adds the field `isSelected` to the rules (aka sets the selected boolean). This new function, `selectedRules` returns only those rules which are selected, if `showOnlySelected` is `true`. The order of these functions doesn't really matter, since `filteredRules` doesn't use `isSelected`.

`selectedRules`, however, is the only function now that requires the `isSelected` boolean to be set, so it calls `addIsSelected` as a dependency.

You may have linked frontend-components incorrectly, perhaps? It works fine for me, and the snapshots seem to be updating as expected in the tests as well. Would you mind trying again please?